Added previousTotalVotesSWOP struc

GubanovDmit opened the pull request.

GubanovDmit committed:
Added previousTotalVotesSWOP struc

GubanovDmit committed:
Update voting.ride

GubanovDmit committed:
Update voting.ride

fpsthirty commented on dApps/SWOP/voting.ride:
```diff
@@ -123,6 +125,14 @@ func votePoolWeight(poolAddressesL:List[String], poolsVoteSWOPNewL:List[Int]) =
     if userPoolVoteSWOPnew < 0 then throw("New userPoolVoteSWOPnew < 0") else
 
+    #entry for FirstHarvest
+    let previousTotalVoteSWOPperiod = this.getStringValue(kPreviousTotalVoteSWOP).split("_")[1].parseIntValue()
```
если в стейте нет ключа kPreviousTotalVoteSWOP (previous_total_struc), то породит ошибку:  "value() called on unit value"; решение: 

```
let previousTotalVoteSWOPperiod = if this.getString(kPreviousTotalVoteSWOP).isDefined()  
	    then this.getStringValue(kPreviousTotalVoteSWOP).split("_")[1].parseIntValue()
	    else currPeriod-1
```

GubanovDmit committed:
Change previous_vote_total to harvest struc

Deleted previous_vote_total and added harvestPoolActiveVoteStruc and harvestUserPoolActiveVoteStruc for farming contract reading

GubanovDmit committed:
Update voting.ride

GubanovDmit committed:
Update voting.ride

throw text edited
